pkg/blockstorage/azure/client.go

```diff
@@ -63,16 +62,12 @@ func NewClient(ctx context.Context, config map[string]string) (*Client, error) {
 		}
 	}
 
-	if cloudEnvironment, ok = config[blockstorage.AzureCloudEnviornmentID]; !ok || cloudEnvironment == "" {
-		cloudEnvironment = azure.PublicCloud.Name
-	}
-
-	env, err := azure.EnvironmentFromName(cloudEnvironment)
+	env, _, err := GetEnvironment(config)
```

Comment:
Not sure about this one because now the reader of this code has to know about reference vs. value types. And unless we update subsequent functions like `getAuthorizer()`, `getCredConfig()` etc. to take `env` to a reference, I am not sure we are gaining much here. 

What's the issue with the previous code?

Reply:
The goal was to bring some information that is decided at this level back to downstream caller. Specifically, information such as `AzureCloudEnviornmentID` or `AzureResurceMgrEndpoint`. 
However, I now think this approach is unnecessarily complicated to achieve that goal. So I have switched to https://github.com/kanisterio/kanister/pull/1466